docs/designs/0001-golang-1-21.md

```diff
@@ -0,0 +1,206 @@
+
+#### Disadvantages:
+- potentially failing builds due to transitively incompatible (i.e. with indirect dependencies) `go` and `toolchain` versions set - but this is a user problem in general though, not cachi2's (mentioning it just in case)
+
+## Decision
+
+Since the [second proposed solution](#2-pre-fetch-all-affecting-toolchains) allows us to both account for the proper support of the toolchains feature, and also to dimish the amount of work needed to support newer versions, the team has opted for it.
+
+## References
+[1] standard Go rules apply when it comes to evaluating modules vs workspaces
+[2] there are some strict rules when it comes to versioning of suggested toolchains which applies transitively too across all dependencies, but luckily that's not our problem to figure out! As noted in https://go.dev/doc/toolchain:
+
+> A module’s go line must declare a version greater than or equal to the go version declared by each of the modules listed in require statements. A workspace’s go line must declare a version greater than or equal to the go version declared by each of the modules listed in use statements.
+
+[3] the `go.env` file has a simple `KEY=VALUE` structure
+[4] Fedora repackaged  _go.env_:
+```
+$ cat /usr/lib/golang/go.env
+
+# This file contains the initial defaults for go command configuration.
+# Values set by 'go env -w' and written to the user's go/env file override these.
+# The environment overrides everything else.
+
+# Use the Go module mirror and checksum database by default.
+# See https://proxy.golang.org for details.
+GOPROXY=direct
+GOSUMDB=off
+
+# Automatically download newer toolchains as directed by go.mod files.
+# See https://go.dev/doc/toolchain for details.
+GOTOOLCHAIN=local
+```
+Go's default vendored _go.env_ :
+```
+# This file contains the initial defaults for go command configuration.
+# Values set by 'go env -w' and written to the user's go/env file override these.
+# The environment overrides everything else.
+
+# Use the Go module mirror and checksum database by default.
+# See https://proxy.golang.org for details.
+GOPROXY=https://proxy.golang.org,direct
+GOSUMDB=sum.golang.org
+
+# Automatically download newer toolchains as directed by go.mod files.
+# See https://go.dev/doc/toolchain for details.
+GOTOOLCHAIN=auto
+```
+
+[5] Snippet from the docs:
+>  toolchain downloads fail for lack of verification if GOSUMDB=off. GOPRIVATE and GONOSUMDB patterns do not apply to the toolchain downloads.
+
+[6] Snippet from the docs:
+> A module cache may be used directly as a file proxy: `GOPROXY=file://$(go env GOMODCACHE)/cache/download`
+
+## Resources
+- https://go.dev/doc/toolchain
+- https://go.dev/ref/mod#environment-variables
```

Comment:
I know I wrote the content, in retrospec ^this should have somehow be part of `References`, no need for another section, admittedly I was trying to be unnecessarily formal, just an FYI.